qmetaobject/src/qtdeclarative.rs

```diff
@@ -24,11 +24,21 @@ use crate::*;
 /// So this is a guard that will be used to panic if the engine is created twice
 static HAS_ENGINE: std::sync::atomic::AtomicBool = std::sync::atomic::AtomicBool::new(false);
 
+#[cfg(feature = "widgets")]
+cpp! {{
+    #include <QtWidgets/QApplication>
+    #define QAPPLICATION QApplication
+}}
+#[cfg(not(feature = "widgets"))]
+cpp! {{
+    #include <QtGui/QGuiApplication>
+    #define QAPPLICATION QGuiApplication
+}}
+
```

Comment:
Hmm, looks like the #[cfg(feature)] macros don't apply to cpp! macro, because both cases seem to be evaluated and QAPPLICATION is always defined as the latter one, regardless of the selected features. Any idea how this could be resolved?